src/config.rs

```diff
@@ -208,6 +208,50 @@ pub trait Config: Sized + Send + Sync + UnwindSafe + RefUnwindSafe + 'static {
         Ok(())
     }
 
+    /// Called upon client connect (if online mode is enabled) to obtain
+    /// the host address of the session server. Defaults to the official
+    /// mojang session server.
+    ///
+    /// This method is called from within the default implementation of
+    /// [`Self::session_server_url`]. If that method is overridden, this
+    /// method will not be called.
+    ///
+    /// # Default Implementation
+    ///
+    /// The official mojang session server (`sessionserver.mojang.com`)
+    /// is used.
+    fn session_server_host(&self, server: &SharedServer<Self>) -> Cow<'_, str> {
+        Cow::from("sessionserver.mojang.com")
+    }
+
+    /// Called upon (every) client connect (if online mode is enabled) to obtain
+    /// the full URL to use for session server requests. Defaults to
+    /// `https://<host>/session/minecraft/hasJoined?username=<username>&serverId=<auth-digest>&ip=<player-ip>`.
+    ///
+    /// If you just want to change the server host, you can override
+    /// [`Self::session_server_host`] instead.
+    ///
+    /// It is assumed, that upon successful request, a structure matching the
+    /// description in the [wiki](https://wiki.vg/Protocol_Encryption#Server) was obtained.
+    /// Providing a URL that does not return such a structure will result in a
+    /// disconnect for every client that connects.
+    ///
+    /// The arguments are described in the linked wiki article.
+    fn format_session_server_url(
+        &self,
+        server: &SharedServer<Self>,
+        username: &str,
+        auth_digest: &str,
+        server_ip: &IpAddr,
```

Comment:
The argument should be called `player_ip`. This can also be passed in by value since it is `Copy`.